internal/controllers/synthesis/slicecleanup.go

```diff
@@ -113,12 +119,13 @@ func synthesisReferencesSlice(syn *apiv1.Synthesis, slice *apiv1.ResourceSlice)
 	return false
 }
 
-func resourcesRemain(slice *apiv1.ResourceSlice) bool {
+func resourcesRemain(comp *apiv1.Composition, slice *apiv1.ResourceSlice) bool {
 	if len(slice.Status.Resources) == 0 && len(slice.Spec.Resources) > 0 {
 		return true // status is lagging behind
 	}
+	shouldOrphan := comp != nil && comp.Annotations != nil && comp.Annotations["eno.azure.io/deletion-strategy"] == "orphan"
 	for _, state := range slice.Status.Resources {
-		if !state.Deleted {
+		if !state.Deleted && !shouldOrphan {
```

Comment:
We no longer set Deleted in the reconciliation controller when orphaning resources, since that doesn't actually make sense. Other callers need to understand orphaning logic now.